lib/src/widgets/board_editor.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:chessground/src/fen.dart';
+import 'package:flutter/widgets.dart';
+
+import 'piece.dart';
+import 'positioned_square.dart';
+import '../models.dart';
+import '../board_editor_settings.dart';
+
+/// A chessboard widget where pieces can be dragged around freely (including dragging piece off and onto the board).
+///
+/// This widget can be used as the basis for a fully fledged board editor, similar to https://lichess.org/editor.
+class BoardEditor extends StatelessWidget {
+  const BoardEditor({
+    super.key,
+    required this.size,
+    required this.orientation,
+    required this.fen,
```

Comment:
You're right that we can replace `String fen` with `Pieces pieces`.

I wanted initially to make the API consistent with `Board`, which uses `fen`. But really it does not make sense to make this conversion between the fen and pieces all the time for the editor. For the board it is different because we need chess logic and it is convenient to export a position through its fen.

I am not entirely satisfied with the global api of chessground. In some parts it uses immutable collections (`IMap`, `ISet`), and in some others mutable ones.
In general I'd use immutable everywhere so I considered using an `IMap<SquareId, Piece>` for the pieces of the `BoardEditor`. But `readFen` returns a regular Map so I think `writeFen` should not deviate from that and I don't want to change `readFen` yet.